pkg/model/firewall.go

```diff
@@ -445,3 +470,102 @@ func generateName(o *awstasks.SecurityGroupRule) string {
 	return fmt.Sprintf("from-%s-%s-%s-%dto%d-%s", src, direction,
 		proto, fi.Int64Value(o.FromPort), fi.Int64Value(o.ToPort), dst)
 }
+
+func (b *FirewallModelBuilder) getExistingRulesFromCloud() ([]*awstasks.SecurityGroupRule, error) {
+	tasks := []*awstasks.SecurityGroupRule{}
+	sgs, err := b.getClusterSecurityGroups()
+	if err != nil {
+		return nil, err
+	}
+	for _, sg := range sgs {
+		name := fi.StringValue(sg.GroupName)
+
+		klog.V(4).Infof("found group %s, %s", name, fi.StringValue(sg.GroupId))
+		// We assume that if the security group name ends with the cluster name it is owned by kOps.
+		// We must ignore security groups about which we don't know anything, as we cannot make tasks of them.
+		if !strings.HasSuffix(name, "."+b.ClusterName()) {
+			klog.V(4).Infof("skipping EC2 security group %q", name)
+			continue
+		}
+		for _, rule := range sg.IpPermissions {
+			ts := b.createRulesFromPermissions(sg, rule, false, sgs)
+			tasks = append(tasks, ts...)
+		}
+		for _, rule := range sg.IpPermissionsEgress {
+			ts := b.createRulesFromPermissions(sg, rule, true, sgs)
+			tasks = append(tasks, ts...)
+		}
+	}
+	return tasks, nil
+}
+
+func (b *FirewallModelBuilder) getClusterSecurityGroups() (map[string]*ec2.SecurityGroup, error) {
+	sgs := make(map[string]*ec2.SecurityGroup)
```

Comment:
Consider moving this line down after line 514, then:
```suggestion
	sgs := make(map[string]*ec2.SecurityGroup, len(response.SecurityGroups))
```